x/concentrated-liquidity/pool_hooks.go

```diff
@@ -108,7 +92,7 @@ func (k Keeper) AfterSwapExactAmountOut(ctx sdk.Context, poolId uint64, sender s
 //
 // Since it is possible for this function to be triggered in begin block code, we need to directly meter its execution and set a limit.
 // If no contract is linked to the hook, this function is a no-op.
-func (k Keeper) callPoolActionListener(ctx sdk.Context, msgBz []byte, poolId uint64, actionPrefix string) (err error) {
+func (k Keeper) callPoolActionListener(ctx sdk.Context, msgBuilderFn func(poolId uint64) ([]byte, error), poolId uint64, actionPrefix string) (err error) {
```

Comment:
It is more usual to put the function parameters as a special type to increase readability
```suggestion
type msgBuildFunc func(poolId uint64) ([]byte, error)

func (k Keeper) callPoolActionListener(ctx sdk.Context, msgBuilder msgBuildFunc, poolId uint64, actionPrefix string) (err error) {
```

Reply:
Good point, ty